Add email subscription form to / and /newsletters

jamesob opened the pull request:
Note that this is based on #18 which should be tested and merged first.

![selection_062](https://user-images.githubusercontent.com/73197/42596224-e5f71c92-8522-11e8-9ca4-0e6500c2a25d.png)

harding commented on newsletters.md:
```diff
@@ -27,3 +27,5 @@ version: 1
   {%- endfor -%}
 </ul>
+
+{% include newsletter-signup.html %}
```
Suggest moving this to the top of the page, probably right after the end of the YAML header (line 11) so it displays  below the Newsletters page heading but above the list.  The reason is that the list of newsletters is going to get longer over time, so this box will move further down the page few people ever see it.

You may also wish to add it to each newsletter (in the same position right below the end of the YAML header), as those pages may receive a lot of incoming link traffic.  There are a few ways to do this, such as creating a layout special to newsletters, putting it in the page layout triggered by a `news_signup: true` option in the YAML header, or just putting the `{% include %}` in each newsletter---the later is the easiest for now, so I'd suggest that. 

Otherwise this LGTM.  I tested and it took me to the mailchimp page and from there back to the live site, but they didn't ask me for double opt-in, so maybe it's in some sort of test mode?

jnewbery commented:
requires rebase on rebased #18.

jnewbery commented:
rebase plz :pray:

jamesob commented:
Rebased and addressed @harding's good feedback. The form is now above the list of newsletters, and I've added a `newsletter` layout that contains a signup form at the bottom of the page.

harding commented:
Tested ACK 41dead2c4d1214d1fbf19a39e656deaa9d940c70, a rather unfortunate commit id, with the caveat that a test of signup doesn't do double opt-in, so I think the list is in test mode or Mailchimp doesn't follow best practices.

jnewbery commented:
> Tested ACK 41dead2, a rather unfortunate commit id

I think Satoshi is trying to communicate with us.

jnewbery commented:
@jamesob is out of office today, so I've gone ahead and pushed a new branch. Changes:

- squashed commits down to two
- changed the CSS slightly so the subscribe area doesn't have a white box surrounding it.
- changed the mailchimp list that the subscribe form adds the subscriber to. Dave - this should fix your comment about double opt-in. Feel free to retest.
- changed the wording from **Get newsletter** to **Subscribe to our newsletter**

harding commented:
Fully tested ACK 258bdbfe665e2f8ee97a26c2d107e88510c4e20a  Thanks!

jnewbery commented:
Thanks @jamesob for code and @harding for review 🌠
